nemo/collections/asr/data/data_simulation.py

```diff
@@ -0,0 +1,1426 @@
+                + file['alignments'][i]
+            )
+
+            if word == "":
+                i += 1
+                continue
+            elif self._text == "":
+                self._text += word
+            else:
+                self._text += " " + word
+            i += 1
+            nw += 1
+            prev_dur_sr = dur_sr
+
+        # add audio clip up to the final alignment
+        if (
+            sentence_duration == 0
+        ) and self._params.data_simulator.session_params.window_type != None:  # cut off the start of the sentence
+            if start_window_amount > 0:  # include window
+                window = self._get_window(start_window_amount, start=True)
+                self._sentence = torch.cat(
+                    (
+                        self._sentence,
+                        np.multiply(audio_file[start_cutoff : start_cutoff + start_window_amount], window),
+                    ),
+                    0,
+                )
+            self._sentence = torch.cat(
+                (self._sentence, audio_file[start_cutoff + start_window_amount : start_cutoff + prev_dur_sr]), 0
+            )
+        else:
+            self._sentence = torch.cat((self._sentence, audio_file[:prev_dur_sr]), 0)
+
+        # windowing at the end of the sentence
+        if (i < len(file['words'])) and self._params.data_simulator.session_params.window_type != None:
+            release_buffer, end_window_amount = self._get_end_buffer_and_window(
+                prev_dur_sr, remaining_duration_sr, len(audio_file[start_cutoff + prev_dur_sr :])
+            )
+            self._sentence = torch.cat(
+                (self._sentence, audio_file[start_cutoff + prev_dur_sr : start_cutoff + prev_dur_sr + release_buffer]),
+                0,
+            )
+            if end_window_amount > 0:  # include window
+                window = self._get_window(end_window_amount, start=False)
+                self._sentence = torch.cat(
+                    (
+                        self._sentence,
+                        np.multiply(
+                            audio_file[
+                                start_cutoff
+                                + prev_dur_sr
+                                + release_buffer : start_cutoff
+                                + prev_dur_sr
+                                + release_buffer
+                                + end_window_amount
+                            ],
+                            window,
+                        ),
+                    ),
+                    0,
+                )
+
+        return sentence_duration + nw, len(self._sentence)
+
+    def _build_sentence(
+        self, speaker_turn: int, speaker_ids: List[str], speaker_lists: List[dict], max_sentence_duration_sr: int
+    ):
+        """
+        Build a new sentences by attaching utterance samples together until the sentence has reached a desired length. 
```

Comment:
Build a new sentence (singular - this is contrary to `build_new_sentences` function)